src/util/errors/UnsupportedError.ts

```diff
@@ -0,0 +1,15 @@
+import { HttpError } from './HttpError';
+
+/**
+ * An error thrown when incoming data is not supported.
+ * Probably because an {@link AsyncHandler} returns false on the canHandle call.
+ */
+export class UnsupportedError extends HttpError {
```

Comment:
Rename to `UnsupportedHttpError`?

Reply:
I thought that might cause confusion since the reason the error is thrown has nothing to do with HTTP, the HTTP link is because internally it stores a status code. But it could make it clearer that it is one of the "Solid errors" that is thrown then so could go either way on this one.

Reply:
But it extends `HttpError`, so it must have something to do with HTTP, right?
If not, then `HttpError` may not be the best name.

Reply:
Oh possibly, I hate naming stuff :D. It's as mentioned in the pull description: it's a way to internally throw errors that have HTTP descriptions in them (i.e. which status code they should return, potentially headers can be added as well later should that be needed), without actually being aware of those HTTP internals (since an `UnsupportedError` does not mention anything HTTP related). So dunno what best to name that.

Reply:
I may be missing something, but I still think this error is HTTP-related, since it clearly includes an HTTP status code. This code would propagate up and will/can be sent to the client, right?

Reply:
Yea I meant internally the class throwing the error does not have to be aware of the HTTP consequences. So that we can easily change the status code that corresponds to this kind of error without the classes throwing it be aware of the change

Reply:
> Yea I meant internally the class throwing the error does not have to be aware of the HTTP consequences.

Shouldn't it? Since the error _is_ HTTP-aware, shouldn't the name reflect it? So that classes throwing the error _are_ aware of it?

Reply:
Well it's mostly a semantics discussion so no idea what it should be. It seemed like a good idea to me to shield the internal classes from the scary HTTP world out there so they could work independently of that. They just get their data, do their stuff and if there's a problem they let it know. The classes calling them should then be responsible for how to handle this to the outside world. I also thought about having some sort of switch cases in the ResponseWriter that converts errors to correct status codes but that seemed less clean and less extensible to me.

Reply:
> It seemed like a good idea to me to shield the internal classes from the scary HTTP world out there so they could work independently of that.

That makes sense, but then the errors should not be related to HTTP (status codes) :-)
When someone uses a class, it should be clear what a class does, so that there's nothing hidden that can lead to unforeseen issues.

> I also thought about having some sort of switch cases in the ResponseWriter that converts errors to correct status codes but that seemed less clean and less extensible to me.

Agreed, that may not be very clean.

Having `Http` in the error class name when a status code has been set seems like a good convention to me :-)
For all other errors, a component higher up can define a status code, or a HTTP 500 error will be thrown if none does.

@RubenVerborgh Do you have any thoughts on this?

Reply:
>For all other errors, a component higher up can define a status code, or a HTTP 500 error will be thrown if none does.

A component higher up defining a status code feels similar to shifting everything to the ResponseWriter. The issue is that several status codes are dependent on specific errors happening, based on the errors described [here](https://github.com/solid/solid-architecture/blob/master/server/request-flow.md). E.g., "`412` if the resource has been modified since the client's last access" will be a specific check somewhere in the ResourceStore.

Or we could just accept that all components are aware of the fact that they are part of an HTTP server I guess.

Reply:
No very strong opinion.

Reply:
Btw, since we might be drifting from the actual issue. I don't have a huge issue with renaming it to UnhandledHttpError if that is the only problem (only seems a bit less clean to me). It's if there are changes requested on how the error handling should be done that I have more opinions ;)